test/mocha/.eslintrc

```diff
@@ -4,6 +4,7 @@
   },
   "globals": {
     "assertNoError": true,
-    "should": true
+    "should": true,
+    "passportStub": true
```

Comment:
I don't think `passportStub` is a global

Reply:
This is the example I was following https://github.com/digitalbazaar/bedrock-account-http/blob/master/test/mocha/.eslintrc#L8.  I tried removing the `passportStub` from here and moving https://github.com/digitalbazaar/bedrock-authn-token-http/pull/13/files#diff-c1129c8b045390789fa8ff62f2c6b4a9R5-R7 into `10-api.js` file, but some of the tests fail when I do that.

Reply:
Here's another example: https://github.com/digitalbazaar/bedrock-profile-http/blob/master/test/mocha/10-api.js#L22 that does not involve a global.

Reply:
@mattcollier On further investigation, it looks like the reason why the stub(for optionallyAuthenticated function) was failing for some of the tests when it is not global and passes when stub is set to global is that some api endpoints has this `optionallyAuthenticated` check  https://github.com/digitalbazaar/bedrock-authn-token-http/blob/master/lib/index.js#L48, while others have `ensureAuthenticated` check
https://github.com/digitalbazaar/bedrock-authn-token-http/blob/master/lib/index.js#L109

The stub works for `ensureAuthenticated` which is an api from brPassport that wraps/calls `optionallyAuthenticated` api https://github.com/digitalbazaar/bedrock-passport/blob/master/lib/index.js#L294 even when it is not set to global.

And doesn’t work for `optionallyAuthenticated` https://github.com/digitalbazaar/bedrock-passport/blob/master/lib/index.js#L273 This seems to be because express keeps a reference to the original `optionallyAuthenticated` function before sinon has a chance to stub it. I think it works for ensureAuthenticated because we don’t stub that directly, but stub optionallyAuthenticated instead. So express keeps a reference to ensureAuthenticated which calls the stub.

The solution to the problem would be either to  set the stub before starting express or make another api in brPassport that would call `optionallyAuthenticated` 

```
	api.foo = async (req, res, next) => {
  		api.optionallyAuthenticated(req, res, next);
	};
```

And replace `optionallyAuthenticated` check with `foo`  in bedrock-auth-token-http

Reply:
Thanks for doing all this research/investigation.  If the stub is setup in test.js it does or does not work?

Reply:
@mattcollier yes, it will work.  I think if we do it before starting bedrock.

Reply:
OK, please do it that way (with the global again if necessary) with a detailed comment that captures all the work you did to explain to future visitors why it's being done in this way.